src/index.js

```diff
@@ -0,0 +1,164 @@
+const imageSize = require('image-size')
+
+module.exports = Joi => {
+  // Validation for width/height parameters
+
+  const dimensionSchema = Joi
+    .number()
+    .positive()
+    .required()
+
+  const typesSchema = Joi
+    .array()
+    .allow(imageSize.types)
+    .min(1)
+    .required()
+
+  return {
+    name: 'image',
+    base: Joi.binary(),
+    language: {
+      minWidth: 'must be at least {{minWidth}} pixels in width',
+      maxWidth: 'must be no more than {{maxWidth}} pixels in width',
+      minHeight: 'must be at least {{minHeight}} pixels in height',
+      maxHeight: 'must be no more than {{maxHeight}} pixels in height',
+      minDimensions: 'must be at least {{minWidth}} pixels in width and {{minHeight}} pixels in height',
+      maxDimensions: 'must be no more than {{minWidth}} pixels in width and {{minHeight}} pixels in height',
+      allowTypes: 'must be one of the following types: {{allowedTypes}}',
+      disallowTypes: 'must be one of the following types: {{allowedTypes}}'
+    },
+
+    coerce (value, state, options) {
```

Comment:
Would it be more appropriate to use `pre` here instead of `coerce`? I seem to recall there being a way/various reasons to turn off coercion and I think `pre` would still run regardless. Mainly thinking for chaining purposes where something else may need to turn off coercion for some reason, say an input form or CLI that specifies dimensions.

Reply:
@randallagordon Indeed, it seems that way—good call.